src/NUnitEngine/nunit.engine/Services/ServiceManager.cs

```diff
@@ -126,6 +126,14 @@ public void StopServices()
             }
         }
 
+        public void SubstituteService<T>(T newService) where T : IService
+        {
+            var currService = GetService(typeof(T));
+            _services.Remove(currService);
+            _serviceIndex.Remove(typeof(T));
+            AddService(newService);
+        }
+
```

Comment:
For test purposes. I'd have ideally made this internal and used InternalsVisibleTo - but as nunit.engine is signed, it wasn't possible without additionally signing nunit-console, and I didn't want to start that cycle off!